cms/djangoapps/contentstore/course_group_config.py

```diff
@@ -84,7 +85,7 @@ def assign_group_ids(self):
         """
         Assign ids for the group_configuration's groups.
         """
-        used_ids = [g.id for p in self.course.user_partitions for g in p.groups]
+        used_ids = [g.id for p in get_course_user_partitions(self.course) for g in p.groups]
```

Comment:
Nit: I wonder if there's a better name for `get_course_user_partitions`?  It was unclear from the name whether this returned only the partitions stored in the course content, or also included the dynamic partitions on the course?
Maybe rename to `get_all_partitions_for_course`?